transport/session/src/types.rs

```diff
@@ -21,21 +22,47 @@ use tracing::{debug, error};
 
 use crate::{errors::TransportSessionError, traits::SendMsg, Capability};
 
+#[cfg(all(feature = "prometheus", not(test)))]
+lazy_static::lazy_static! {
+    static ref METRIC_SESSION_INNER_SIZES: hopr_metrics::MultiHistogram =
+        hopr_metrics::MultiHistogram::new(
+            "hopr_session_inner_sizes",
+            "Sizes of data chunks fed from inner session to HOPR protocol",
+            vec![20.0, 40.0, 80.0, 160.0, 320.0, 640.0, 1280.0],
+            &["session_id"]
+    ).unwrap();
+}
+
+// Enough to fit Ed25519 peer IDs and 2-byte tag number
+const MAX_SESSION_ID_STR_LEN: usize = 64;
+
 /// Unique ID of a specific session.
 ///
 /// Simple wrapper around the maximum range of the port like session unique identifier.
 /// It is a simple combination of an application tag and a peer id that will in future be
 /// replaced by a more robust session id representation.
-#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[derive(Clone, Copy)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct SessionId {
     tag: u16,
     peer: PeerId,
+    // Since this SessionId is commonly represented as a string,
+    // we cache its string representation here.
+    // Also by using a statically allocated ArrayString, we allow the SessionId to remain Copy.
+    // This representation is possibly truncated to MAX_SESSION_ID_STR_LEN.
+    cached: arrayvec::ArrayString<MAX_SESSION_ID_STR_LEN>,
 }
 
 impl SessionId {
     pub fn new(tag: u16, peer: PeerId) -> Self {
-        Self { tag, peer }
+        let mut cached = format!("{peer}:{tag}");
+        cached.truncate(MAX_SESSION_ID_STR_LEN);
+
+        Self {
+            tag,
+            peer,
+            cached: cached.parse().expect("cannot fail due to truncation"),
+        }
```

Comment:
The string representation of SessionId is always ASCII, since PeerIDs are represented as Base58 string.
The 58 in the name tells you that it never leaves the ASCII range, thank you for your dilligence.